File: common/src/main/java/com/utilities/swipeback/ISwipeBack.java
package com.utilities.swipeback;

import com.utilities.views.SwipeBackLayout;

/**
 * Created by slai4 on 4/17/2016.
 */
public interface ISwipeBack {
    /**
     * @return the SwipeBackLayout associated with this activity.
     */
    SwipeBackLayout getSwipeBackLayout();

    void setSwipeBackEnable(boolean enable);

    /**
     * Scroll out contentView and finish the activity
     */
    void scrollToFinishActivity();

}

File: common/src/main/java/com/utilities/swipeback/ISwipeBack.kt
package com.utilities.swipeback

import com.utilities.views.SwipeBackLayout

/**
 * Created by slai4 on 4/17/2016.
 */
interface ISwipeBack {
    /**
     * @return the SwipeBackLayout associated with this activity.
     */
    val swipeBackLayout: SwipeBackLayout?

    fun setSwipeBackEnable(enable: Boolean)

    /**
     * Scroll out contentView and finish the activity
     */
    fun scrollToFinishActivity()
}
